File: medicloud/medicloud-patients/medicloud-patients-domain/src/main/java/com/jclinical/patients/domain/model/Patient.java
package com.jclinical.patients.domain.model;

import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Builder
@ToString
public class Patient {
    private final UUID id;
    private final UUID clinicId;
    private final String firstName;
    private final String lastNamePaterno;
    private final String lastNameMaterno;
    private final String curp;
    private final LocalDate dateOfBirth;
    private final Gender gender;
    private final String phone;
    private final String email;
    private final String occupation;
    private final MaritalStatus maritalStatus;
    private final String nationality;
    private final BloodType bloodType;
    private final Address address;
    private final EmergencyContact emergencyContact;
    private final LocalDateTime createdAt;
    private final LocalDateTime updatedAt;
}

File: medicloud/medicloud-frontend/src/types/patient.ts
export type Gender = "MALE" | "FEMALE" | "OTHER";

export type MaritalStatus = "SINGLE" | "MARRIED" | "DIVORCED" | "WIDOWED" | "COHABITATION";

export type BloodType =
  | "O_POSITIVE"
  | "O_NEGATIVE"
  | "A_POSITIVE"
  | "A_NEGATIVE"
  | "B_POSITIVE"
  | "B_NEGATIVE"
  | "AB_POSITIVE"
  | "AB_NEGATIVE";

export interface Address {
  street?: string;
  outdoorNumber?: string;
  indoorNumber?: string;
  colonia?: string;
  municipality?: string;
  state?: string;
  zipCode?: string;
}

export interface EmergencyContact {
  fullName?: string;
  relationship?: string;
  phone?: string;
}

export interface PatientResponse {
  id: string;
  clinicId: string;
  firstName: string;
  lastNamePaterno: string;
  lastNameMaterno?: string;
  curp?: string;
  dateOfBirth?: string;
  gender?: Gender;
  phone?: string;
  email?: string;
  occupation?: string;
  maritalStatus?: MaritalStatus;
  nationality?: string;
  bloodType?: BloodType;
  address?: Address;
  emergencyContact?: EmergencyContact;
  createdAt: string;
  updatedAt: string;
}

export interface RegisterPatientRequest {
  clinicId: string;
  firstName: string;
  lastNamePaterno: string;
  lastNameMaterno?: string;
  curp?: string;
  dateOfBirth?: string;
  gender?: Gender;
  phone?: string;
  email?: string;
  occupation?: string;
  maritalStatus?: MaritalStatus;
  nationality?: string;
  bloodType?: BloodType;
  address?: Address;
  emergencyContact?: EmergencyContact;
}

export interface UpdatePatientRequest {
  firstName: string;
  lastNamePaterno: string;
  lastNameMaterno?: string;
  curp?: string;
  dateOfBirth?: string;
  gender?: Gender;
  phone?: string;
  email?: string;
  occupation?: string;
  maritalStatus?: MaritalStatus;
  nationality?: string;
  bloodType?: BloodType;
  address?: Address;
  emergencyContact?: EmergencyContact;
}
